src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -701,6 +751,11 @@ private static object ProcessDictionary(IDictionary dict, int depth, in ConvertT
 
             foreach (DictionaryEntry entry in dict)
             {
+                if (entry.Value == null && context.IgnoreNullValue)
+                {
+                    continue;
+                }
```

Comment:
There is this bug also: https://github.com/JamesNK/Newtonsoft.Json/issues/1997

Reply:
Looking at the issue you mentioned it seems that the Newtonsoft behavior is by design and not a bug at all. But should it work this way? Why?

Reply:
Core behavior is the same. I think we should follow Newtonsoft and Core.

Reply:
Let me see if I got it right: what keeps the serializer `NullValueHandling.Ignore` from working properly is the fact that we convert everything into dictionaries and for dictionaries and arrays the behavior is to always keep nulls.

So we need the workaround for types that were not originally dicts or arrays: JObject, PSObject and CustomObject. Maybe we don't need it at all for JObject, I'm not sure why it was being converted into a dict in the first place.

Reply:
Yes, you got it right. I guess it come from limitations of old Core APIs. MSFT team used Newtonsoft library but custom serialization at porting time. (Perhaps it is not easy to implement a converter on Newtonsoft.Json).
In last days I found that .Net Core 3.0 has new power Json API and started working on porting  ConvertFrom-Json from Netonsoft.Json to Core. After some experiments I had to switch to ConvertTo-Json cmdlet. I was supposed to do this work after merging the PR, but now I’m thinking of pulling my PR because it will simplifying the code and adding IgnoreNullValue - I hope it will as simple as adding an option.

Reply:
Since this will simplify a lot the work I will wait until the port is complete and add IgnoreNullVale in the new code